sound/soc/sof/intel/hda-dai.c

```diff
@@ -452,9 +458,24 @@ int sdw_hda_dai_hw_params(struct snd_pcm_substream *substream,
 	if (!hext_stream)
 		return -ENODEV;
 
-	/* in the case of SoundWire we need to program the PCMSyCM registers */
+	/*
+	 * in the case of SoundWire we need to program the PCMSyCM registers. In case
+	 * of aggregated devices, we need to define the channel mask for each sublink
+	 * by reconstructing the split done in soc-pcm.c
+	 */
+	for_each_rtd_cpu_dais(rtd, i, dai) {
+		if (dai == cpu_dai)
+			break;
+	}
```

Comment:
Can we use other than `i`, like `cpu_dai_id`? It is not obvious right away that we only need the `i` form this loop later.

Reply:
ok

Reply:
done